src/commonControls/questionnaireControl/QuestionnaireControl.ts

```diff
@@ -0,0 +1,1578 @@
+            // SingleValueControlIntents that mention question (target) and answer (feedback slot)
+            name: 'FeedbackAnswerToSpecificQuestion (builtin)',
+            canHandle: this.isFeedbackAnswerToSpecificQuestion,
+            handle: this.handleFeedbackAnswerToSpecificQuestion,
+        },
+        {
+            name: 'SpecificAnswerByTouch (builtin)',
+            canHandle: this.isSpecificAnswerByTouch,
+            handle: this.handleSpecificAnswerByTouch,
+        },
+        {
+            name: 'CompletionRequestByVoice (builtin)',
+            canHandle: this.isCompletionRequestByVoice,
+            handle: this.handleCompletionRequest,
+        },
+        {
+            name: 'CompletionRequestByTouch (builtin)',
+            canHandle: this.isCompletionRequestByTouch,
+            handle: this.handleCompletionRequest,
+        },
+        {
+            name: 'BareYesToCompletionQuestion (builtin)',
+            canHandle: this.isBareYesToCompletionQuestion,
+            handle: this.handleCompletionRequest,
+        },
+        {
+            name: 'BareNoToCompletionQuestion (builtin)',
+            canHandle: this.isBareNoToCompletionQuestion,
+            handle: this.handleBareNoToCompletionQuestion,
+        },
+    ];
+
+    // tsDoc - see Control
+    async canHandle(input: ControlInput): Promise<boolean> {
+        const stdHandlers = this.standardInputHandlers;
+        const customHandlers = this.props.inputHandling.customHandlingFuncs ?? [];
+
+        const matches = [];
+        for (const handler of stdHandlers.concat(customHandlers)) {
+            if (await handler.canHandle.call(this, input)) {
+                matches.push(handler);
+            }
+        }
+
+        if (matches.length > 1) {
+            log.error(
+                `More than one handler matched. Handlers in a single control should be mutually exclusive. ` +
+                    `Defaulting to the first. handlers: ${JSON.stringify(matches.map((x) => x.name))}`,
+            );
+        }
+
+        if (matches.length >= 1) {
+            this.handleFunc = matches[0].handle.bind(this);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // tsDoc - see Control
+    async handle(input: ControlInput, resultBuilder: ControlResultBuilder): Promise<void> {
+        const content = this.getQuestionnaireContent(input);
+        if (this.handleFunc === undefined) {
+            log.error(
+                'QuestionnaireControl: handle called but this.handlerFunc not set.  are canHandle/handle out of sync?',
+            );
+            throw new Error(`this.handlerFunc not set.  are canHandle/handle out of sync?`);
+        }
+
+        await this.handleFunc(input, resultBuilder);
+
+        this.state.activeInitiative = undefined; // clear the initiative state so we don't get confused on subsequent turns.
+
+        //NOTE: this is a trial of _not_ adding initiative during handle.. but rather
```

Comment:
In this control, the handlers never add initiative.  It works out fine and should be a better in general as there are likely some situations where user-initiative causes a control to handle something but the initiative has to return to where they were.  So I'm preparing for that eventual feature.
I think it will also be a simplification to the programming model to say that initiative can only come from the initiative phase and will prepare everyone for the complicated case mentioned.